### DISCUSSION TITLE
Group License online verification

### DISCUSSION THREAD

**Discussion by SethXie** [dashingsoft/pyarmor#1366] [Ideas]
Currently Pyarmor Group supports Offline obfuscation and other types of license require online verify license.

In addition to this, is that possible to support online license verification for Pyarmor Group too, using the same command as others?

`$ pyarmor reg pyarmor-regfile-xxxx.zip`

By enabling this, we don't have to create a separate group device file for each machine, as most of the time those machines can access the Internet.

**Comment by jondy** [comment]
Sorry, group license is designed for offline obfuscation.

**Reply by SethXie** [reply]
@jondy Thanks for your reply. Understood, but I expected Group license to have stronger features instead of more restrictions than other licenses. The problem that we're facing is that we use a docker runner to build our applications, and the runner may be destoryed at the end of its job. In this case, it's complicated if we need to register the device (runner) every time. What's worse is that we may very likely hit the 100 limitation within 24 hours, because:
> In 24 hours only less than 100 devices can use one same Pyarmor License

Questions:

1. Is there any limitation of total number of devices? In another word, can we have 10000 registered machines but less than 100 are used within 24 hours?
2. Do you have any plan for group license to support online obfuscation **in the future**?
3. When would Pyarmor 8.3 (unlimited docker) be ready?
4. Do you have any other suggestions to deal with this case?

**Comment by jondy** [comment]
Sorry, group license is designed for offline obfuscation.

**Comment by jondy** [comment]
Hi, unlimited docker container has been supported in Pyarmor 8.2.9 for group license, please check
https://pyarmor.readthedocs.io/en/stable/how-to/register.html#run-unlimited-dockers-in-offline-device

At this time there is no plan that group license will support online verification because it changes a lot in design.

At this time only 100 offline devices available for one group license.

**Reply by SethXie** [reply]
Fair enough. Thanks for your reply!
